### PR TITLE
[DEVHAS-134] Sanitize Git error messages

### PR DESCRIPTION
**Pull request by johnmcollier**
Fixes https://issues.redhat.com/browse/DEVHAS-134

Adds a new util function (+ tests) for sanitizing error messages. Can expand in the future as needed. Updates the Component controller to call this function before reporting errors from the gitops library.

### CODE REVIEW

File: controllers/component_controller.go [hunk 1]
```diff
@@ -475,8 +475,9 @@ func (r *ComponentReconciler) generateGitops(ctx context.Context, component *app
 
 	err = gitops.GenerateAndPush(tempDir, remoteURL, *component, r.Executor, r.AppFS, gitOpsBranch, gitOpsContext, gitopsConfig)
 	if err != nil {
-		log.Error(err, "unable to generate gitops resources due to error")
-		return err
+		gitOpsErr := util.SanitizeErrorMessage(err)
```
Comment (maysunfaisal) [inline controllers/component_controller.go:478]: needs to be also called on `RemoveAndPush()`

Comment (johnmcollier) [reply controllers/component_controller.go:478]: Good catch

Comment (johnmcollier) [reply controllers/component_controller.go:478]: Fixed

**Comment by maysunfaisal** [comment]
@johnmcollier I wonder if we should as part of a diff issue sanitize this more broadly.. For instance, 

1. directly call the util fn in *_controller_conditions.go
2. create a log wrapper, to filter out any possible future confidential info

this way we can maintain a list of regex(s) and not worry about where to call it 🤔

**Comment by johnmcollier** [comment]
@maysunfaisal Yes, agreed.

I do some concerns around the overhead of too many regex checks on _every_ log statement, but we can look into that

**Review by maysunfaisal — APPROVED** [review]

**Pull request merged**